kernel/src/log_segment/tests.rs

```diff
@@ -1428,3 +1440,157 @@ fn test_compaction_starts_at_checkpoint() {
         None,    // version to load
     );
 }
+
+enum ExpectedFile {
+    Commit(Version),
+    Compaction(Version, Version),
+}
+
+fn test_commit_cover(
+    commit_versions: &[u64],
+    compaction_versions: &[(u64, u64)],
+    checkpoint_version: Option<u64>,
+    version_to_load: Option<u64>,
+    expected_files: &[ExpectedFile],
+) {
+    let log_segment = create_segment_for(
+        commit_versions,
+        compaction_versions,
+        checkpoint_version,
+        version_to_load,
+    );
+    let cover = log_segment.find_commit_cover();
+    // our test-utils include "_delta_log" in the path, which is already in log_segment.log_root, so
+    // we don't use them. TODO: Unify this
+    let expected_locations = expected_files.iter().map(|ef| match ef {
+        ExpectedFile::Commit(version) => log_segment
+            .log_root
+            .join(&format!("{version:020}.json"))
+            .expect("Couldn't join"),
+        ExpectedFile::Compaction(lo, hi) => log_segment
+            .log_root
+            .join(&format!("{lo:020}.{hi:020}.compacted.json"))
+            .expect("Couldn't join"),
+    });
+    assert_eq!(cover.len(), expected_locations.len());
+    for (location, expected_location) in cover.iter().zip(expected_locations) {
+        assert_eq!(location.location, expected_location);
+    }
+}
+
+#[test]
+fn test_commit_cover_one_compaction() {
```

Comment:
should we include tests for (just off the top of my head trying to think through coverage):
- no compaction files
- 'touching' compactions (like end = beginning)?

Reply:
yep, added